src/nginx/t/rollout_config_download_no_config_id.t

```diff
@@ -0,0 +1,286 @@
+#
+use strict;
+use warnings;
+use JSON::PP;
+
+################################################################################
+
+use src::nginx::t::ApiManager;    # Must be first (sets up import path to
+                                  # the Nginx test module)
+use src::nginx::t::HttpServer;
+use src::nginx::t::ServiceControl;
+use Test::Nginx;    # Imports Nginx's test module
+use Test::More;     # And the test framework
+use Data::Dumper;
+
+################################################################################
+
+# Port assignments
+my $NginxPort          = ApiManager::pick_port();
+my $BackendPort        = ApiManager::pick_port();
+my $ServiceControlPort = ApiManager::pick_port();
+my $ServiceManagementPort = ApiManager::pick_port();
+my $MetadataPort = ApiManager::pick_port();
+
+my $t = Test::Nginx->new()->has(qw/http proxy/)->plan(11);
+
+# Save servce configuration that disables the report cache.
+# Report request will be sent for each client request
+$t->write_file('server.pb.txt', <<"EOF");
+service_control_config {
+  report_aggregator_config {
+    cache_entries: 0
+    flush_interval_ms: 1000
+  }
+}
+metadata_server_config: {
+  enabled: true
+  url: "http://127.0.0.1:${MetadataPort}"
+}
+service_management_config {
+  url: "http://127.0.0.1:${ServiceManagementPort}"  
+}
+EOF
+
+ApiManager::write_file_expand( $t, 'nginx.conf', <<"EOF");
+%%TEST_GLOBALS%%
+daemon off;
+error_log /tmp/error.log debug;
+
+events {
+  worker_connections 32;
+}
+http {
+  %%TEST_GLOBALS_HTTP%%
+  server_tokens off;
+  server {
+    listen 127.0.0.1:${NginxPort};
+    server_name localhost;
+    location / {
+      endpoints {
+        server_config server.pb.txt;
+        %%TEST_CONFIG%%
+        on;
+      }
+      proxy_pass http://127.0.0.1:${BackendPort};
+    }
+  }
+}
+EOF
+
+my $report_done = 'report_done';
+
+$t->run_daemon( \&bookstore, $t, $BackendPort, 'bookstore.log' );
+$t->run_daemon( \&servicecontrol, $t, $ServiceControlPort,
+    'servicecontrol.log', $report_done);
+$t->run_daemon( \&servicemanagement, $t, $ServiceManagementPort,
+    'servicemanagement.log');
+$t->run_daemon( \&metadata, $t, $MetadataPort, 'metadata.log');
+
+is( $t->waitforsocket("127.0.0.1:${BackendPort}"), 1, 'Bookstore socket ready.');
+is( $t->waitforsocket("127.0.0.1:${ServiceControlPort}"), 1, 
+    'Service control socket ready.' );
+is( $t->waitforsocket("127.0.0.1:${ServiceManagementPort}"), 1, 
+    'Service management socket ready.' );
+is( $t->waitforsocket("127.0.0.1:${MetadataPort}"), 1, 
+    'Metadata socket ready.' );
+
+$t->run();
+
+################################################################################
+
+my $response = ApiManager::http_get(
+  $NginxPort, '/shelves?key=this-is-an-api-key' );
+
+$t->stop_daemons();
+
+my ( $response_headers, $response_body ) = split /\r\n\r\n/, $response, 2;
+
+like( $response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.' );
```

Comment:
please add comment
# config loading fail due to no config_id,  
# the policy is fail open, all requests will go with loading failure.